### ISSUE TITLE
match patterns should support uconstr

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#9321]
#### Version

8.8.2

#### Description of the problem

This code should work, but it does not:
```coq
Goal True.
  let v := uconstr:(True) in
  lazymatch constr:(v) with
  | v => idtac
  end. (* Error: No matching clauses for match. *)
```

I'd also like this code to work:
```coq
Goal True.
  let v := uconstr:(_) in
  lazymatch constr:(True) with
  | v => idtac
  end. (* Error: No matching clauses for match. *)
```

cc @ppedrot 

See Also #4425

**Comment by ppedrot** [comment]
My personal belief is that uconstr is a horrible hack, and we shouldn't recommend its use. Furthermore, this particular example is using another well-known half-broken part of Ltac pattern-matching, namely matching against a constr variable (and the infamous `pattern_of_constr`). Thus I wouldn't hold my breath on that being fixed.

**Comment by maximedenes** [comment]
@ppedrot should we close this as a wontfix, then?

**Comment by ppedrot** [comment]
@maximedenes I'd say personally so.

**Comment by maximedenes** [comment]
@JasonGross @toineo I suggest you try to port your code to Ltac2, and open an issue if you encounter the same kind of problems.

**Comment by JasonGross** [comment]
@maximedenes @ppedrot I'd be more enthusiastic about porting to Ltac2 if
(a) Ltac2 was shipped with Coq by default, like the ssr tactics are
(b) Ltac2 got `Message.printf ` or some other satisfactory replacement for `idtac` during debugging, and
(c) Ltac2 would awake from the stasis it seems to be in; I haven't seen much response to issues in the past couple of months, and @ppedrot's most recent non-merge commit is from November

**Comment by ppedrot** [comment]
@JasonGross I haven't had much time for development since around the beginning of December, but this is hopefully going to change soon. Also, Ltac2 is probably going to be merged back into Coq for 8.10 assuming we work around the stdlib-related limitations of the build system. As for `printf`, that's pretty easy to implement using a dedicated parsing scope and an phantom type-annotated payload datatype.
